sources/packages/api/src/route-controllers/models/common.dto.ts

```diff
@@ -15,14 +15,30 @@ export class AddressAPIInDTO {
   addressLine2: string;
   @IsNotEmpty()
   city: string;
-  @IsNotEmpty()
+  @IsOptional()
   provinceState: string;
   @IsNotEmpty()
   country: string;
   @IsNotEmpty()
   postalCode: string;
+  @IsOptional()
+  // This property will have canada postal code.
+  // This property is for formIO purpose.
+  @IsOptional()
+  canadaPostalCode?: string;
```

Comment:
Can we check the possibility to use the conditional validators here: https://github.com/typestack/class-validator#conditional-validation.

Reply:
I like this conditional validator. actually, I next story is the continuation of this, in the story, I will be validating the form using dry run and I may replace this validators with @Allow()